packages/wallet/backend/tests/transaction/service.test.ts

```diff
@@ -0,0 +1,224 @@
+import { createApp, TestApp } from '@/tests/app'
+import { Knex } from 'knex'
+import { truncateTables } from '@/tests/tables'
+import type { AuthService } from '@/auth/service'
+import { mockedListAssets, mockedTransactionInsertObjs } from '../mocks'
+import { TransactionService } from '@/transaction/service'
+import { Transaction } from '@/transaction/model'
+import { faker } from '@faker-js/faker'
+import { loginUser } from '@/tests/utils'
+import { Account } from '@/account/model'
+import { PaymentPointer } from '@/paymentPointer/model'
+
+describe('Transaction Controller', (): void => {
+  let bindings: Container<Bindings>
+  let appContainer: TestApp
+  let knex: Knex
+  let authService: AuthService
+  let transactionService: TransactionService
+  let userId: string
+
+  // "dependency" = ALL foreign keys (paymentPointers, account, user...)
+  const prepareTransactionDependencies = async () => {
+    const account = await Account.query().insert({
+      name: faker.string.alpha(10),
+      userId,
+      assetCode: mockedListAssets[0].code,
+      assetId: mockedListAssets[0].id,
+      assetScale: mockedListAssets[0].scale,
+      virtualAccountId: 'mocked'
+    })
+
+    const paymentPointer = await PaymentPointer.query().insert({
+      url: faker.string.alpha(10),
+      publicName: faker.string.alpha(10),
+      accountId: account.id,
+      id: faker.string.uuid()
+    })
+
+    return {
+      account,
+      paymentPointer
+    }
+  }
+
+  beforeAll(async (): Promise<void> => {
+    bindings = createContainer(env)
+    appContainer = await createApp(bindings)
+    knex = appContainer.knex
+    authService = await bindings.resolve('authService')
+    transactionService = await bindings.resolve('transactionService')
+  })
+
+  beforeEach(async (): Promise<void> => {
+    const extraUserArgs = {
+      isEmailVerified: true,
+      rapydWalletId: 'mocked'
+    }
+
+    const { user } = await loginUser({
+      authService,
+      extraUserArgs
+    })
+    userId = user.id
+  })
+
+  afterAll(async (): Promise<void> => {
+    appContainer.stop()
+    knex.destroy()
+  })
+
+  afterEach(async (): Promise<void> => {
+    await truncateTables(knex)
+  })
+
+  describe('listAll', (): void => {
+    it('should list all transactions (0 transactions)', async (): Promise<void> => {
+      const transactions = await transactionService.listAll({
+        userId,
+        paginationParams: {
+          page: 0,
+          pageSize: Infinity
+        },
+        filterParams: {},
+        orderByDate: 'asc'
+      })
+
+      expect(transactions.total).toEqual(0)
+      expect(transactions.results.length).toEqual(0)
+    })
+
+    it('should list all transactions (4 transactions)', async (): Promise<void> => {
+      const { paymentPointer, account } = await prepareTransactionDependencies()
+
+      await Promise.all(
+        mockedTransactionInsertObjs.map(async (mockedTransactionInsertObj) =>
+          Transaction.query().insert({
```

Comment:
insert is working with arrays as well, so the transactions list can be passed in one go